app/connectors/CompanyHouseConnector.scala

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package connectors
+
+import config.FrontendAppConfig
+import models.CompanyDetails
+import play.api.Logger
+import play.api.http.Status._
+import uk.gov.hmrc.http.logging.Authorization
+import uk.gov.hmrc.http.{HeaderCarrier, HttpClient, HttpReads, HttpResponse}
+import uk.gov.hmrc.play.HeaderCarrierConverter
+
+import javax.inject.Inject
+import scala.concurrent.{ExecutionContext, Future}
+
+class CompanyHouseConnector @Inject()(appConfig: FrontendAppConfig, http: HttpClient) {
+
+  val logger = Logger(classOf[CompanyHouseConnector])
+
+  implicit val httpReads: HttpReads[HttpResponse] = new HttpReads[HttpResponse] {
+    override def read(method: String, url: String, response: HttpResponse) = response
+  }
+  def getName(response: HttpResponse): String ={
+    (response.json \ "company_name").as[String].toLowerCase.replace(" ","")
+  }
+  def validateCRN(data: CompanyDetails)(implicit ec: ExecutionContext): Future[Option[Boolean]] = {
+    val hc = HeaderCarrier().withExtraHeaders(("Authorization", appConfig.companyHouseRequestAuth))
+    http.GET(appConfig.companyHouseRequestUrl + data.companyReferenceNumber)(httpReads, hc:HeaderCarrier, ec:ExecutionContext)
+      .map{response =>
```

Comment:
```suggestion
      .map { response =>
```
There's a couple more scalastyle kinda things in this file that would make this even more readable/consistent 
i.e. Spaces before and after ```=```
Spaces after a ```:``` when giving a type definition etc.